presto-rcfile/src/main/java/com/facebook/presto/rcfile/binary/TimestampEncoding.java

```diff
@@ -54,7 +56,7 @@ public void encodeColumn(Block block, SliceOutput output, EncodeOutput encodeOut
     @Override
     public void encodeValueInto(Block block, int position, SliceOutput output)
     {
-        writeTimestamp(output, type.getLong(block, position));
+        writeTimestamp(output, microsToMillis(type.getLong(block, position)));
```

Comment:
Are we truncating timestamps here? Why?

Reply:
Yes, this change is made to be compatible with existing milliseconds logic，which remove micros precision before processing and pad 0 to convert to micros at last.